integrationTests/multiShard/smartContract/scCallingSC_test.go

```diff
@@ -18,10 +20,233 @@ import (
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	idxProposers := make([]int, numOfShards+1)
+	for i := 0; i < numOfShards; i++ {
+		idxProposers[i] = i * nodesPerShard
+	}
+	idxProposers[numOfShards] = numOfShards * nodesPerShard
+
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Node.Stop()
+		}
+	}()
+
+	initialVal := big.NewInt(1000000000)
+	integrationTests.MintAllNodes(nodes, initialVal)
+
+	firstSCOwner := nodes[0].OwnAccount.Address.Bytes()
+
+	// deploy the smart contracts
+	firstSCAddress := putDeploySCToDataPool("./testdata/counter.wasm", firstSCOwner, 0, big.NewInt(50), nodes)
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+	integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
+	integrationTests.SyncBlock(t, nodes, idxProposers, round)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	// make smart contract call to shard 1 which will do in shard 0
+	for _, node := range nodes {
+		txData := "increment"
+		for i := 0; i < 10; i++ {
+			integrationTests.CreateAndSendTransaction(node, big.NewInt(0), firstSCAddress, txData)
+		}
+	}
+
+	time.Sleep(time.Second)
+
+	nrRoundsToPropagateMultiShard := 7
+	for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
+		integrationTests.UpdateRound(nodes, round)
+		integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
+		integrationTests.SyncBlock(t, nodes, idxProposers, round)
+		round = integrationTests.IncrementAndPrintRound(round)
+		nonce++
+	}
+
+	address, _ := integrationTests.TestAddressConverter.CreateAddressFromPublicKeyBytes(firstSCAddress)
+	shId := nodes[0].ShardCoordinator.ComputeId(address)
+	for _, node := range nodes {
+		if node.ShardCoordinator.SelfId() != shId {
+			continue
+		}
+
+		numCalled := vm.GetIntValueFromSC(nil, node.AccntState, firstSCAddress, "get", nil)
+		require.NotNil(t, numCalled)
+	}
+
+	account := getAccountFromAddrBytes(nodes[0].AccntState, nodes[0].OwnAccount.Address.Bytes())
+	require.Equal(t, big.NewInt(0), account.DeveloperReward)
+
+	newOwnerAddress := []byte("12345678123456781234567812345678")
+	txData := "ChangeOwnerAddress" + "@" + hex.EncodeToString(newOwnerAddress)
+	integrationTests.CreateAndSendTransaction(nodes[0], big.NewInt(0), firstSCAddress, txData)
+
+	for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
+		integrationTests.UpdateRound(nodes, round)
+		integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
+		integrationTests.SyncBlock(t, nodes, idxProposers, round)
+		round = integrationTests.IncrementAndPrintRound(round)
+		nonce++
+	}
+
+	// check new owner address is set
+	account = getAccountFromAddrBytes(nodes[0].AccntState, firstSCAddress)
+	require.Equal(t, newOwnerAddress, account.OwnerAddress)
+	require.True(t, account.DeveloperReward.Cmp(big.NewInt(0)) == 1)
```

Comment:
could replace with `require.Greater(t, account.DeveloperReward.Uint64(), uint64(0))`

Reply:
i will let it as it is